lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -361,10 +361,6 @@ defmodule Kernel.ErrorsTest do
     assert_compile_fail CompileError,
       "nofile:1: unknown key :age for struct Kernel.ErrorsTest.GoodStruct",
       '%#{GoodStruct}{age: 27} = %{}'
-
-    assert_compile_fail CompileError,
-      "nofile:1: unknown key ^field for struct Kernel.ErrorsTest.GoodStruct",
-      '%#{GoodStruct}{^field => 27} = %{}'
```

Comment:
So you should define the variable before hand. We should put in expansion_test. I believe there is a `describe "structs"` in there.